src/olympia/files/models.py

```diff
@@ -182,18 +182,21 @@ def generate_hash(self, filename=None):
         with open(filename or self.current_file_path, 'rb') as fobj:
             return f'sha256:{get_sha256(fobj)}'
 
-    def generate_filename(self, extension=None):
+    def generate_filename(self):
         """
         Files are in the format of:
         {addon_name}-{version}-{apps}
         (-{platform} for some of the old ones from back when we had multiple
          platforms)
+
+        By convention, newly signed files after 2022-03-31 get a .xpi
+        extension, unsigned get .zip. This helps ensure CDN cache is busted
+        when we sign something.
```

Comment:
how much of a concern are files that were uploaded prior to 2022-03-31, so have an `.xpi` extension even though they aren't signed, but may be then signed?  I understand it may not be an immediate problem until there other changes to how files are served but will there be a point where it is?  And if so is there a migration strategy for those older files?

Reply:
I've just added that to https://github.com/mozilla/addons-server/issues/11510 - I don't think it's a big concern but let's tackle that during the big file directory structure migration.